### PR TITLE
Add a caching HTTP client

### PR DESCRIPTION
**Pull request by Kegsay**
Most of the RSS feeds I `curl`'d had HTTP cache headers (sometimes to say, don't cache at all, other times to put a low `max-age`), so a generic RFC-compliant HTTP caching library will be Good Enough for our use cases. E.g:

```
> GET /news/rss.xml?edition=uk HTTP/1.1
> Host: feeds.bbci.co.uk
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Server: nginx
< Content-Type: text/xml; charset=utf-8
< X-Content-Type-Options: nosniff
< x-origin-route: xrt-ext
< Content-Length: 28853
< X-Cache-Action: HIT
< X-Cache-Hits: 2
< X-Cache-Age: 8
< Cache-Control: max-age=51
< Date: Tue, 11 Oct 2016 14:38:24 GMT
< Connection: keep-alive
<
```

I verified that the caching library is doing its job via `tcpdump`, and indeed it seems to be!

### CODE REVIEW

File: src/github.com/matrix-org/go-neb/services/rssbot/rssbot.go [hunk 1]
```diff
@@ -214,6 +220,7 @@ func itemToHTML(feed *gofeed.Feed, item gofeed.Item) matrix.HTMLMessage {
 }
 
 func init() {
+	cachingClient = httpcache.NewMemoryCacheTransport().Client()
```
Comment (NegativeMjark) [inline src/github.com/matrix-org/go-neb/services/rssbot/rssbot.go]: Are there any limits on how much memory this cache will consume?

**Comment by NegativeMjark** [comment]
LGTM

File: src/github.com/matrix-org/go-neb/services/rssbot/rssbot.go (hunk 1 above)
Comment (Kegsay) [inline src/github.com/matrix-org/go-neb/services/rssbot/rssbot.go]: No, though it looks like we can plug-and-play an [LruCache](https://github.com/die-net/lrucache).

Comment (Kegsay) [inline src/github.com/matrix-org/go-neb/services/rssbot/rssbot.go]: Done.

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**